pkg/purl/purl.go

```diff
@@ -198,13 +198,25 @@ func (p *PackageURL) BOMRef() string {
 	return purl.String()
 }
 
+// NewPackageIdentifier creates a new package identifier based on PURL for the given package.
+func NewPackageIdentifier(target ftypes.TargetType, pkg ftypes.Package) ftypes.PkgIdentifier {
```

Comment:
This function is tested indirectly.
Can we add small unit test for it?

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/5e72c5a95058110fdaeac484791abac7d35a288e